### PR TITLE
feat: add `gid` field support for exec* family

### LINKED ISSUE #2144: Probes don't export updated `gid` information for `execve` system call
**Describe the bug**

For `execve` system calls, probes correctly export the effective user ID as `uid`. However, they don't export the effective group ID, which can change when the user run a set-group-ID program.

**How to reproduce it**

Run the `execve` system call on a binary having the set-group-ID bit set and belonging to a group different from the current user's one. 

**Expected behaviour**

The probes correctly exports the effective group id and Falco uses this information to update the internal state for the process.

**Screenshots**

**Environment**

- Falco version:

`0.39.1`
- System info: 

```
{
  "machine": "x86_64",
  "nodename": "ekoops-XPS-15-9530",
  "release": "6.8.0-48-generic",
  "sysname": "Linux",
  "version": "#48~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Mon Oct  7 11:24:13 UTC 2"
}
```
- Cloud provider or hardware configuration:
- OS:

```
PRETTY_NAME="Ubuntu 22.04.5 LTS"
NAME="Ubuntu"
VERSION_ID="22.04"
VERSION="22.04.5 LTS (Jammy Jellyfish)"
VERSION_CODENAME=jammy
ID=ubuntu
ID_LIKE=debian
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
UBUNTU_CODENAME=jammy
```
- Kernel:

`Linux ekoops-XPS-15-9530 6.8.0-48-generic #48~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Mon Oct  7 11:24:13 UTC 2 x86_64 x86_64 x86_64 GNU/Linux`
- Installation method:

From source.
**Additional context**

### PR DESCRIPTION
**Pull request by ekoops**
**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

/kind bug

> /kind cleanup

> /kind design

> /kind documentation

> /kind failing-test

/kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area API-version

> /area build

> /area CI

/area driver-kmod

/area driver-bpf

/area driver-modern-bpf

> /area libscap-engine-bpf

> /area libscap-engine-gvisor

> /area libscap-engine-kmod

> /area libscap-engine-modern-bpf

> /area libscap-engine-nodriver

> /area libscap-engine-noop

> /area libscap-engine-source-plugin

> /area libscap-engine-savefile

> /area libscap

> /area libpman

/area libsinsp

/area tests

> /area proposals

**Does this PR require a change in the driver versions?**

> /version driver-API-version-major

> /version driver-API-version-minor

> /version driver-API-version-patch

> /version driver-SCHEMA-version-major

/version driver-SCHEMA-version-minor

> /version driver-SCHEMA-version-patch

**What this PR does / why we need it**:
This PR adds the effective group ID to the set of fields exported by the `execve` and `execveat` system calls. In this way, it is possible to detect an effective group ID change upon set-group-ID programs execution.
**Which issue(s) this PR fixes**:

Fixes #2144 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
feat(drivers): add `gid` field for exec* family
```

### CODE REVIEW

**Review by Andreagit97 — COMMENTED** [review]
Great work! Just one question

File: test/drivers/event_class/event_class.cpp [hunk 1]
```diff
@@ -1081,12 +1081,7 @@ void event_test::assert_cgroup_param(int param_num) {
 		        << VALUE_NOT_CORRECT << m_current_param;
 	}
 
-	/* With the kmod we send more cgroups than the 5 we send in bpf and modern bpf */
-	if(is_kmod_engine()) {
-		assert_param_len_ge(total_len);
-	} else {
-		assert_param_len(total_len);
-	}
+	assert_param_len_ge(total_len);
```
[the remark this answers is not in the document]
Comment (ekoops) [reply test/drivers/event_class/event_class.cpp:1084]: To be honest I don't remember the exact reason. It should be related to a test failing locally. Anyway, I'm going to revert it for now.

**Review by Andreagit97 — APPROVED** [review]
/approve

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**